src/components/Products/Filters/Price.js

```diff
@@ -0,0 +1,88 @@
+import React, { useState, useEffect } from 'react';
+import { useNavigate, useLocation } from 'react-router-dom';
+
+const Price = () => {
+  const location = useLocation();
+  const navigate = useNavigate();
+
+  let searchLine = location.search;
+  const urlParams = new URLSearchParams(searchLine);
+  const minPriceParam = urlParams.get('minPrice');
+  const maxPriceParam = urlParams.get('maxPrice');
+
+  const [minPrice, setMinPrice] = useState(minPriceParam || '');
+  const [maxPrice, setMaxPrice] = useState(maxPriceParam || '');
+
+  const queryStart = () => {
+    return searchLine.length ? '&' : '?';
+  };
+
+  useEffect(() => {
+    const price = setTimeout(() => {
+      if (+minPrice) {
+        if (!minPriceParam) {
```

Comment:
I already wrote you function for prices. Why are these double checks here again?

Reply:
Because I want the filters to be flexible in any use case.